### PR TITLE
tools(build-tools): bump eslint-config-fluid to 5.1.0

### PR DESCRIPTION
**Pull request by jason-ha**
fix defects and remove unused disables

### CODE REVIEW

File: build-tools/packages/build-cli/src/library/package.ts [hunk 1]
```diff
@@ -68,9 +68,7 @@ export async function npmCheckUpdates(
 	depsToUpdate: ReleasePackage[] | RegExp[],
 	releaseGroupFilter: ReleaseGroup | undefined,
 	depUpdateType: DependencyUpdateType,
-	// eslint-disable-next-line default-param-last
```
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/library/package.ts:71]: Was this rule removed?

File: build-tools/packages/build-cli/src/library/git.ts [hunk 2]
```diff
@@ -8,7 +8,6 @@ import readPkgUp from "read-pkg-up";
 import { SimpleGit, SimpleGitOptions, simpleGit } from "simple-git";
 
 // type-fest seems to trigger this lint rule, which seems to be a false positive.
```
Comment (tylerbutler) [inline build-tools/packages/build-cli/src/library/git.ts:10]: Remove this comment as well.

**Review by tylerbutler — CHANGES_REQUESTED** [review]

File: build-tools/packages/build-cli/src/library/package.ts (hunk 1 above)
Comment (jason-ha) [reply build-tools/packages/build-cli/src/library/package.ts:71]: Not with this update. Those disables were unused prior to this update. The removals were applied by running lint with --fix.
The rule is not so relevant to TypeScript anyway - tsc will complain if there are required parameters without defaults after optional/defaulted parameters.

**Review by tylerbutler — APPROVED** [review]

**Comment by tylerbutler** [comment]
@jason-ha just an FYI for you - one reason I disabled the autofixer for unused disables is twofold:

1. Before pnpm, we had problems with dep upgrades sneaking into the repo, and disables would disappear either due to false negative bugs or other eslint changes. Then that issue would go away and lint would start failing. This is no longer relevant with pnpm IMO.
2. The autofixer for disable comments leaves a blank line (or it used to) Before prettier, this was really annoying. Maybe prettier helps address it.

Not sure these are the most relevant today, just thought you might be wondering why the fix command excludes these fixes in the client release group.

**Comment by jason-ha** [comment]
> @jason-ha just an FYI for you - one reason I disabled the autofixer for unused disables is twofold:
> 
> 1. Before pnpm, we had problems with dep upgrades sneaking into the repo, and disables would disappear either due to false negative bugs or other eslint changes. Then that issue would go away and lint would start failing. This is no longer relevant with pnpm IMO.
> 2. The autofixer for disable comments leaves a blank line (or it used to) Before prettier, this was really annoying. Maybe prettier helps address it.
> 
> Not sure these are the most relevant today, just thought you might be wondering why the fix command excludes these fixes in the client release group.

Thanks - good info to have. It would be nice to have unused disables be errors.
The auto-fixer left a line with a space. It was annoying. I didn't see if prettier would handle them because I was reviewing each change anyway. Probably auto-fix+prettier gets close to the best result.

**Pull request merged**